modules/nf-core/abra2/realign/main.nf

```diff
@@ -0,0 +1,60 @@
+
+process ABRA2_REALIGN {
+    tag "$meta.id" + (binding.variables["meta2"] ? ",$meta2.id" : ",none") + (binding.variables["meta3.id"] ? ",$meta3.id" : ",none")
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::abra2=2.24" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/quay.io/biocontainers/abra2:2.24--h9f5acd7_1':
+        'quay.io/biocontainers/abra2:2.24--h9f5acd7_1' }"
+
+    input:
+    tuple val(meta), path(bam),path(bam_bai)
+    tuple val(meta2), path(bam2),path(bam_bai2)
+    tuple val(meta3), path(bam3),path(bam_bai3)
+    tuple path(fasta), path(fasta_fai)
+    path(targets)
+
+    output:
+    tuple val(meta), path("${meta.id}.abra.bam"), emit: bam
```

Comment:
Does the tool always need a maximum of 3 BAMs? I think it would maybe be better if a list of BAMs was used as input instead of separate input channels for every BAM

Reply:
The issue is that the bams need to be tracked in and out.  For example, we use it to realign tumor and normal bam together, and it's important that the tumor.abra.bam and normal.abra.bam aren't swapped.  It's not just taking an arbitrary list of bams and their relative metadata is important.

Reply:
Will this be the case for every use case of the tool? (if so, then it's fine by me how you implement it now)

Reply:
We use it in 3 ways.  alone, with T-N DNA bams, and with T-N DNA, and an RNA bam.  It might be okay to implement a bam-list-input version for use cases where we don't need to worry about the sample metadata, but that wouldn't work for the cases. we actually use in practice.

Reply:
This is an interesting case, could you maybe post this on Slack in the #modules channel?